src/components/MagicCodeInput.tsx

```diff
@@ -155,17 +152,18 @@ function MagicCodeInput(props) {
         clear() {
             lastFocusedIndex.current = 0;
             setInputAndIndex(0);
-            inputRefs.current.focus();
-            props.onChangeText('');
+            inputRefs.current?.focus();
+            onChangeTextProp('');
         },
         blur() {
             blurMagicCodeInput();
         },
     }));
 
     const validateAndSubmit = () => {
-        const numbers = decomposeString(props.value, props.maxLength);
-        if (wasSubmitted || !props.shouldSubmitOnComplete || _.filter(numbers, (n) => ValidationUtils.isNumeric(n)).length !== props.maxLength || props.network.isOffline) {
+        const numbers = decomposeString(value, maxLength);
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        if (wasSubmitted || !shouldSubmitOnComplete || numbers.filter((n) => ValidationUtils.isNumeric(n)).length !== maxLength || isOffline) {
```

Comment:
Have you tested the `isOffline` value there and if it works the same way as before?